storage-access.bs

```diff
@@ -155,25 +158,31 @@ When invoked on {{Document}} |doc|, the <dfn export method for=Document><code>ha
+1. If |doc|'s [=Document/browsing context=] is same authority with |doc|'s [=Document/browsing context=]'s [=top-level browsing context=], [=/resolve=] |p| with true and return |p|.
+
+ISSUE: "same authority" here is a placeholder for a future concept that allows user agents to perform [=same site=] checks while adhering to additional security aspects such as the presence of a cross-site parent document, see [whatwg/storage#142](https://github.com/whatwg/storage/issues/142#issuecomment-1122147159). For the purpose of implementation, the user agent should use this step to determine whether the embedded document does not need to invoke {{Document/requestStorageAccess()}} to be granted storage access. This may involve comparing the [=site for cookies=] or performing a [=same site=] check with the top-level document.
+
 1. [=Queue a global task=] on the [=permissions task source=] given |global| to [=/resolve=] |p| with |global's| [=environment/has storage access=].
 1. Return |p|.
 
-ISSUE: Shouldn't step 8 be [=same site=]?
-
 When invoked on {{Document}} |doc|, the <dfn export method for=Document><code>requestStorageAccess()</code></dfn> method must run these steps:
 
 <!-- https://developer.mozilla.org/en-US/docs/Web/API/Document/requestStorageAccess -->
 
 1. Let |p| be [=a new promise=].
 1. If |doc| is not [=Document/fully active=], then [=/reject=] |p| with an "{{InvalidStateError}}" {{DOMException}} and return |p|.
 1. Let |global| be |doc|'s [=relevant global object=].
+1. Let |settings| be |doc|'s [=relevant settings object=].
 1. If |global| is not a [=secure context=], then [=/reject=] |p| with a "{{NotAllowedError}}" {{DOMException}} and return |p|.
 1. If |doc|'s [=Document/browsing context=] is a [=top-level browsing context=], [=/resolve=] and return |p|.
 1. If |doc| is not [=allowed to use=] "`storage-access`", [=/reject=] |p| with a "{{NotAllowedError}}" {{DOMException}} and return |p|.
 1. If |doc|'s [=Document/origin=] is an [=opaque origin=], [=/reject=] |p| with a "{{NotAllowedError}}" {{DOMException}} and return |p|.
-1. If the [=top-level origin=] of |doc|'s [=relevant settings object=] is an [=opaque origin=], [=/reject=] |p| with a "{{NotAllowedError}}" {{DOMException}} and return |p|.
-1. If |doc|'s [=Document/origin=] is [=same origin=] with the [=top-level origin=] of |doc|'s [=relevant settings object=], [=/resolve=] and return |p|.
+1. If |settings|'s [=top-level origin=] is an [=opaque origin=], [=/reject=] |p| with a "{{NotAllowedError}}" {{DOMException}} and return |p|.
+1. Let |topLevelSite| be the result of [=obtain a site|obtaining a site=] from |settings|' [=top-level origin=].
+1. Let |embeddedSite| be the result of [=obtain a site|obtaining a site=] from |settings|' [=environment settings object/origin=].
+1. If |embeddedSite| is [=same site=] with |topLevelSite|:
```

Comment:
We should explain why this does not need to be same authority. (Or fix it, I haven't thought too hard about it.)

Reply:
It doesn't, because we still want to auto-grant for same-site (intentionally ignoring security aspects of authority). I can add a note.

Reply:
PTAL :)